geode-core/src/distributedTest/java/org/apache/geode/internal/cache/PartitionedRegionClearDUnitTest.java

```diff
@@ -171,26 +183,146 @@ private void verifyCacheListenerTriggerCount(MemberVM serverVM) {
     }
   }
 
+  SerializableCallableIF<Integer> getWriterClears = () -> {
+    int clears =
+        clearsByRegion.get(REGION_NAME) == null ? 0 : clearsByRegion.get(REGION_NAME).get();
+    return clears;
+  };
+
+  SerializableCallableIF<Integer> getWriterDestroys = () -> {
+    int destroys =
+        destroysByRegion.get(REGION_NAME) == null ? 0 : destroysByRegion.get(REGION_NAME).get();
+    return destroys;
+  };
+
+  void configureServers(boolean dataStoreWithWriter, boolean accessorWithWriter) {
+    dataStore1.invoke(() -> initDataStore(true, dataStoreWithWriter));
+    dataStore2.invoke(() -> initDataStore(true, dataStoreWithWriter));
+    dataStore3.invoke(() -> initDataStore(true, dataStoreWithWriter));
+    accessor.invoke(() -> initAccessor(true, accessorWithWriter));
+    // make sure only datastore3 has cacheWriter
+    dataStore1.invoke(() -> {
+      Region region = getRegion(false);
+      region.getAttributesMutator().setCacheWriter(null);
+    });
+    dataStore2.invoke(() -> {
+      Region region = getRegion(false);
+      region.getAttributesMutator().setCacheWriter(null);
+    });
+  }
+
+  @Test
+  public void normalClearFromDataStoreWithWriterOnDataStore() {
+    configureServers(true, true);
+    client1.invoke(this::initClientCache);
+    client2.invoke(this::initClientCache);
+
+    accessor.invoke(() -> feed(false));
+    verifyServerRegionSize(NUM_ENTRIES);
+    dataStore3.invoke(() -> getRegion(false).clear());
+    verifyServerRegionSize(0);
+    verifyCacheListenerTriggerCount(dataStore3);
+
+    assertThat(dataStore1.invoke(getWriterClears)).isEqualTo(0);
+    assertThat(dataStore2.invoke(getWriterClears)).isEqualTo(0);
+    assertThat(dataStore3.invoke(getWriterClears)).isEqualTo(1);
+    assertThat(accessor.invoke(getWriterClears)).isEqualTo(0);
+
+    dataStore3.invoke(() -> {
+      Region region = getRegion(false);
+      region.destroyRegion();
```

Comment:
The test is doing region destroy but the name suggests it is for clear. All the tests are doing region destroy, rather than clear.

Reply:
I use destroy to compare. I can add a comments in the case. If I removed the destroy part, it's hard to show what's the expected behavior.

Reply:
Probably separating test may be good idea. One for destroy and one for Clear.